src/Components/DetailItem/DetailItem.js

```diff
@@ -0,0 +1,206 @@
+import React from "react";
+import { withRouter } from "react-router-dom";
+import { ip } from "./../../Pages/Login/ip";
+import "./DetailItem.scss";
+
+class DetailItem extends React.Component {
+  state = {
+    input: 1,
+    data: {},
+    heart: "",
+  };
+
+  calculationHandler = (e) => {
+    let { input } = this.state;
+    const { name } = e.target;
+    const isPlus = name === "plus";
+
+    if (!isPlus && input === 1) return 1;
+
+    this.setState({
+      input: isPlus ? (input += 1) : (input -= 1),
+    });
+  };
+
+  quantityHandler = (e) => {
+    this.setState({
+      input: e.target.value,
+    });
+  };
+
+  componentDidMount = () => {
+    fetch(`${ip}/product/${this.props.match.params.id}`, {
+      method: "GET",
+    })
+      .then((res) => res.json())
+      .then((res) => {
+        this.setState({
+          data: res.data,
+        });
+      });
+  };
+
+  shoppingHandler = () => {
+    localStorage.setItem("cart_count", this.state.input);
+
+    fetch(`http://${ip}/cart/add`, {
+      method: "POST",
+      headers: {
+        Authorization: localStorage.getItem("token"),
+      },
+      body: JSON.stringify({
+        product: this.state.data.id,
+        quantity: this.state.input,
+      }),
+    });
+
+    this.props.history.push("/Cart");
+  };
+
+  cardPrice = () => {
+    return (this.state.data.price * 2) / 100;
+  };
+
+  heartHandler = () => {};
+
+  decimalRemove = (data) => {
+    return Math.floor(data).toLocaleString(undefined, {
+      maximumFractionDigits: 5,
+    });
+  };
+
+  render() {
+    const { input, data } = this.state;
```

Comment:
```suggestion
    const { input, data: {
      brand_logo, brand, brand_desc 등등등
    } } = this.state;
```
이런 식으로 한 층 더 구조분해 할당 할 수 있습니다

Reply:
와우 이런방법이..?

Reply:
수정완료